mmv1/third_party/terraform/fwtransport/framework_config.go.tmpl

```diff
@@ -520,6 +523,26 @@ func (p *FrameworkProviderConfig) SetupGrpcLogging() {
 	)
 }
 
+// Remove the `/{{version}}/` from a base path if present.
+func RemoveBasePathVersion(url string) string {
+	re := regexp.MustCompile(`(?P<base>http[s]://.*)(?P<version>/[^/]+?/$)`)
+	return re.ReplaceAllString(url, "$1/")
+}
+
+func (p *FrameworkProviderConfig) NewIamCredentialsClient(userAgent string) *iamcredentials.Service {
+	iamCredentialsClientBasePath := RemoveBasePathVersion(p.IAMCredentialsBasePath)
+	log.Printf("[INFO] Instantiating Google Cloud IAMCredentials client for path %s", iamCredentialsClientBasePath)
+	clientIamCredentials, err := iamcredentials.NewService(p.Context, option.WithHTTPClient(p.Client))
+	if err != nil {
+		log.Printf("[WARN] Error creating client iam credentials: %s", err)
+		return nil
+	}
+	clientIamCredentials.UserAgent = userAgent
+	clientIamCredentials.BasePath = iamCredentialsClientBasePath
+
+	return clientIamCredentials
+}
+
```

Comment:
Please move this into fwtransport/framework_provider_clients.go alongside `func (p *FrameworkProviderConfig) NewDnsClient`!